frappe/public/js/frappe/router.js

```diff
@@ -349,8 +348,6 @@ frappe.router = {
 	push_state(url) {
 		// change the URL and call the router
 		if (window.location.pathname !== url) {
-			// cleanup any remenants of v1 routing
-			window.location.hash = '';
```

Comment:
Removing this line fixes unnecessary `hashchange` triggers and unexpected behaviors in certain cases while going back.